### PR TITLE
Fixes #28935 - Add mutations for media

### PR DESCRIPTION
**Pull request by xprazak2**

### CODE REVIEW

File: app/graphql/mutations/media/update.rb [hunk 1]
```diff
@@ -0,0 +1,10 @@
+module Mutations
+  module Media
+    class Update < UpdateMutation
+      graphql_name 'UpdateMediumMutation'
+      description 'Updates a new installation medium'
+
```
Comment (kamils-iRonin) [inline app/graphql/mutations/media/update.rb:6]: I think we need `id` here, maybe `Mutations::UpdateMutation` is a good place to define it?
```suggestion
argument :id, ID, required: true
```

File: test/graphql/mutations/media/update_mutation_test.rb [hunk 2]
```diff
@@ -0,0 +1,109 @@
+require 'test_helper'
+
+module Mutations
+  module Models
+    class UpdateMutationTest < ActiveSupport::TestCase
+      let(:medium) { FactoryBot.create(:medium) }
+      let(:medium_id) { Foreman::GlobalId.for(medium_id) }
```
Comment (kamils-iRonin) [inline test/graphql/mutations/media/update_mutation_test.rb]:
```suggestion
      let(:medium_id) { Foreman::GlobalId.for(medium) }
```

File: test/graphql/mutations/media/update_mutation_test.rb [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+      let(:os_id) { Foreman::GlobalId.for(FactoryBot.create(:operatingsystem)) }
+      let(:org_id) { Foreman::GlobalId.for(FactoryBot.create(:organization)) }
+      let(:loc_id) { Foreman::GlobalId.for(FactoryBot.create(:location)) }
+      let(:variables) do
+        {
```
Comment (kamils-iRonin) [inline test/graphql/mutations/media/update_mutation_test.rb:12]: I think `id` is missing

```suggestion
        {
          id: medium_id,
```

File: test/graphql/mutations/media/update_mutation_test.rb [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+          locationIds: [loc_id],
+        }
+      end
+      let(:query) do
+        <<-GRAPHQL
+        mutation UpdateMediumMutation(
+            $id: ID!,
+            $name: String!,
+            $path: String!,
+            $osFamily: String,
```
Comment (kamils-iRonin) [inline test/graphql/mutations/media/update_mutation_test.rb]: `OsFamilyEnum` is the correct type for `osFamily`

```suggestion
            $osFamily: OsFamilyEnum,
```

File: app/graphql/mutations/media/update.rb (hunk 1 above)
Comment (shiramax) [reply app/graphql/mutations/media/update.rb]: :+1:

**Review by shiramax — CHANGES_REQUESTED** [review]
Looks good to me!
can you please add more negative tests? 
- update a media without id 
- create media without all the required fields
- create media with  invalid valid os family
- create media with a name that is already exists
 
also, what about querying of getting index or show information of specific media.

**Comment by xprazak2** [comment]
I added more tests. We already have queries for [index and show](https://github.com/theforeman/foreman/blob/develop/app/graphql/types/query.rb#L86-L87)

**Review by shiramax — APPROVED** [review]

**Comment by shiramax** [comment]
thanks @xprazak2 ! merging! also thanks @kamils-iRonin  for reviewing!

**Pull request merged**